src/output/plugins/httpd/HttpdClient.cxx

```diff
@@ -41,6 +47,10 @@ HttpdClient::BeginResponse() noexcept
 {
 	assert(state != State::RESPONSE);
 
+	if (!head_method) {
+	       should_reject_unauthorized = (strlen(httpd.password) > 0 && (!provided_password || (strcmp(provided_password->c_str(), httpd.password) != 0)));
```

Comment:
It is [advised](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#compare-password-hashes-using-safe-functions) to use constant-time string comparison for such cases to avoid [timing attacks](https://en.wikipedia.org/wiki/Timing_attack). Has this been considered?

Reply:
Given that the password is transmitted without any encryption, it would be a lot easier to just capture the plain password from the net than to do timing-based guesses, so no, constant-time string comparisons would be pointless overengineering here.

Reply:
If you *really* want, I have no problem with implementing it in constant-time, but as the password is stored in config file in plaintext, and transmitted over unencrypted connection, I think it's indeed a bit too much.